packages/zowe-explorer/src/trees/job/JobTree.ts

```diff
@@ -853,6 +854,14 @@ export class JobTree extends ZoweTreeProvider<IZoweJobTreeNode> implements Types
                 await SettingsConfig.setDirectValue(JobTree.persistenceSchema, setting);
             }
         }
+        if (e.affectsConfiguration(Constants.SETTINGS_JOBS_DEFAULT_SORT)) {
+            const sortOpts = SharedUtils.getDefaultSortOptions(JobUtils.JOB_SORT_OPTS, Constants.SETTINGS_JOBS_DEFAULT_SORT, Sorting.JobSortOpts);
+            for (const sessionNode of this.mSessionNodes) {
+                sessionNode.sort = sortOpts;
+                this.sortBy(sessionNode);
+            }
```

Comment:
I think I've spotted where the desync occurs. After changing default sort order, this works to update the default sort for _non-favorited_ session nodes, but does not account for favorited nodes. We'll need to add some logic here (likely need a helper function that we can call here for the sake of unit tests) that traverses through favorites and updates the default sorting value for the favorited entries